src/lab/Task8.java

```diff
@@ -0,0 +1,364 @@
+package lab;
+
+import java.util.Scanner;
+
+public class Task8 {
+    /**
+     * TASk 2, TASK 3, and TASK 4, and TASK 5, and Task 6, Task 7, Task 8 IMPLEMENTATION.
+     * Some of task2, task3, and task4, task5, task6, task 7 parts may have been changed when writing task7
+     **/
+
+    static Movie[] copyArray(Movie[] movies, int size) {
+        if (movies.length == 0) {
+            return new Movie[size];
+        } else {
+            Movie[] tempArr = new Movie[size];
+            System.arraycopy(movies, 0, tempArr, 0, movies.length);
+            return tempArr;
+        }
+    }
+
+    static void insertMovie(Scanner sc, Movie[] movies, int movieArraySize) {
+        for (int i = movieArraySize; i < movies.length; i++) {
+            System.out.println("""
+                    Enter the movie name: """);
+            String movieName = sc.next();
+            System.out.println("Enter the movie rating: ");
+            double movieRating = sc.nextDouble();
+            Movie movie = new Movie(movieName, movieRating);
+            movies[i] = movie;
+        }
+    }
+
+    static String findMaxOrMin(Movie[] movies, boolean findMax) {
+        double movieRating = movies[0].getRating();
+        int movieIndex = 0;
+        for (int i = 0; i < movies.length; i++) {
+            if ((findMax && movies[i].getRating() > movieRating) ||
+                    (!findMax && movies[i].getRating() < movieRating)) {
+                movieRating = movies[i].getRating();
+                movieIndex = i;
+            }
+        }
+        return movies[movieIndex].getName();
+
+    }
+
+    static double findAverage(Movie[] movies) {
```

Comment:
and you can find average in loop which you also find max and min, inside one loop you can get these three values